security/certificate_types_descriptions/machine-config-operator-certificates.adoc

```diff
@@ -8,24 +8,48 @@ toc::[]
 
 == Purpose
 
-Machine Config Operator certificates are used to secure connections between the Red Hat Enterprise Linux CoreOS (RHCOS) nodes and the Machine Config Server.
+This certificate authority is used to secure connections from nodes to Machine Config Server (MCS) during initial provisioning.
+
+There are two certificates:
+. A self-signed CA, the MCS CA
+. A derived certificate, the MCS cert
+
+=== Provisioning details
+
+{product-title} installations using {op-system-first} are installed using Ignition. This process is split into two parts:
+
+. An Ignition config is created which is references a URL for the full configuration served by the MCS.
+. The Ignition config manifests as a `worker.ign` file created by the `openshift-install` command for user-provisioned infrastucture installation methods. For installer-provisioned infrastructure installation methods that use the Machine API Operator, this configuration appears as the `worker-user-data` secret.
 
 include::snippets/mcs-endpoint-limitation.adoc[]
 
 .Additional resources
 
+* xref:../../post_installation_configuration/machine-configuration-tasks.adoc#understanding-the-machine-config-operator[Understanding the Machine Config Operator].
+
 * xref:../../networking/openshift_sdn/about-openshift-sdn.adoc#about-openshift-sdn[About the OpenShift SDN network plugin].
 
+=== Provisioning chain of trust
+
+The MCS CA is injected into the Ignition configuration under the `security.tls.certificateAuthorities` configuration field. The MCS then provides the complete configuration with a webserver using the MCS cert.
+
+The client, Ignition running on a machine in the initramfs, validates that the MCS cert presented by the server has a chain of trust to an authority, the MCS CA, it recognizes. This ensures that the client is accessing the correct server.
```

Comment:
I'm really not sure what the first sentence on line 36 means. Also, should "initramfs" be in monospace font? I've seen that term referenced as a file name before.

Reply:
@lahinson I am trying to describe using the client/server architecture how certificates get validated and used. Is there a particular part that's unclear? It is kind of a mouthful. Is this rewording any more clear:

> The client validates that the MCS cert presented by the server has a chain of trust to an authority, the MCS CA, it recognizes. This ensure that the client is a    ccessing the correct server. The client in this case is Ignition running on a machine in the initramfs.

I think in this situation it should remain as is. I'm using it like it's used here: https://docs.openshift.com/container-platform/4.13/architecture/architecture-rhcos.html#ignition-sequence_architecture-rhcos. in some cases initramfs is used to refer to the specific collection of files for installation, `initramfs` files, and that's when i see the term monospaced...I think these situations are different

Reply:
@jldohmann Thanks for the explanation! Your revision is better, although I wonder if the end of the first sentence can be clearer. Would something like this work?

>The client validates that the MCS cert that the server presented has a chain of trust to an authority that the MCS cert recognizes, such as the MCS CA. This ensures that the client is accessing the correct server. In this case, the client is Ignition running on a machine in the initramfs.

Reply:
> The client validates that the MCS cert that the server presented has a chain of trust to an authority that the MCS cert recognizes, such as the MCS CA. 

It's the other way around, the MCS CA signs the MCS cert.  I feel my initial wording here was reasonably accurate for what it's worth.

See also the "security.tls.certificateAuthorities" section under https://coreos.github.io/ignition/configuration-v3_4/ which is how this works - the MCS CA is included in that value.

Reply:
I recommend a glance at https://en.wikipedia.org/wiki/Certificate_authority btw

Reply:
@cgwalters Thanks for the clarification and the resources. In the initial wording that I reviewed, which I've included below, I had a hard time figuring out what the two phrases that are surrounded by commas ("Ignition running on a machine in the initramfs" and "the MCS CA") were modifying. 

>The client, Ignition running on a machine in the initramfs, validates that the MCS cert presented by the server has a chain of trust to an authority, the MCS CA, it recognizes.

Jesse's suggested revision clarified the meaning of the first phrase. However, as I'm sure you could tell based on my last comment, I'm still having trouble understanding how the phrase "the MCS CA" fits in. 

I appreciate your help to get this right. Is this revision more accurate?

>The MCS CA signs the MCS cert. The client uses the MCS cert to authenticate the signature and validates that the MCA cert that the server presents has a chain of trust to a recognized authority.

Reply:
I've updated the wording a little bit

Reply:
LTGM! Thanks for your patience in considering all of this feedback.